gronx.go

```diff
@@ -96,11 +96,44 @@ func Segments(expr string) ([]string, error) {
 // SegmentsDue checks if all cron parts are due.
 // It returns bool. You should use IsDue(expr) instead.
 func (g *Gronx) SegmentsDue(segs []string) (bool, error) {
-	for pos, seg := range segs {
+	skipMonthDayCheck := false
+	for i := 0; i < len(segs); i++ {
```

Comment:
does the presence/absence of year 7th seg cause any issue with reverse looping? i think not

Reply:
Yes I did that at first, but in this case position 0 would be either year or day of week, and all further positions would need a check on segments size first. Since a lot of the existing logic bases itself on the position (switch case for bumping, iteration limits for each segment, also position is passed as a param to a lot of function, checks for weekday or month day in validator), it would require to do a check on the segment size in a lot of places.

I figured it would be cleaner to keep your segments identifiers intact by just reverting the iteration order without changing them.